integrations/observability/grafana/resources.yaml

```diff
@@ -0,0 +1,220 @@
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: observability
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: grafana-lgtm
+  namespace: observability
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: grafana-lgtm
+  template:
+    metadata:
+      labels:
+        app: grafana-lgtm
+    spec:
+      containers:
+        - name: grafana-lgtm
+          image: grafana/otel-lgtm:latest
+          ports:
+            - containerPort: 3000
+            - containerPort: 4317
+            - containerPort: 4318
+            - containerPort: 9411
+          volumeMounts:
+            - name: otelcol-config
+              mountPath: /otel-lgtm/otelcol-config.yaml
+              subPath: otelcol-config.yaml
+      volumes:
+        - name: otelcol-config
+          configMap:
+            name: otelcol-config
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: grafana-lgtm
+  namespace: observability
+spec:
+  selector:
+    app: grafana-lgtm
+  ports:
+    - name: grafana
+      port: 3000
+    - name: otel-grpc
+      port: 4317
+    - name: otel-http
+      port: 4318
+    - name: zipkin
+      port: 9411
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: otelcol-config
+  namespace: observability
+data:
+  otelcol-config.yaml: |
+    ---
+    receivers:
+      otlp:
+        protocols:
+          grpc:
+            endpoint: 0.0.0.0:4317
+          http:
+            endpoint: 0.0.0.0:4318
+      prometheus/collector:
+        config:
+          scrape_configs:
+            - job_name: 'opentelemetry-collector'
+              static_configs:
+                - targets: ['localhost:8888']
+      zipkin:
+        endpoint: 0.0.0.0:9411
+
+    processors:
+      batch:
+
+    exporters:
+      otlphttp/metrics:
+        endpoint: http://localhost:9090/api/v1/otlp
+        tls:
+          insecure: true
+      otlphttp/traces:
+        endpoint: http://localhost:4418
+        tls:
+          insecure: true
+      otlphttp/logs:
+        endpoint: http://localhost:3100/otlp
+        tls:
+          insecure: true
+
+    service:
+      pipelines:
+        traces:
+          receivers: [otlp, zipkin]
+          processors: [batch]
+          exporters: [otlphttp/traces]
+        metrics:
+          receivers: [otlp,prometheus/collector]
+          processors: [batch]
+          exporters: [otlphttp/metrics]
+        logs:
+          receivers: [otlp]
+          processors: [batch]
+          exporters: [otlphttp/logs]
```

Comment:
Please consider using kustomize with a configmapgenerator for extracting this to an external file

Reply:
Right, it will be much better. Done.